src/backend/controllers/battleController.js

```diff
@@ -0,0 +1,267 @@
+// @flow strict
+
+const { validationResult } = require('express-validator');
+
+import type { Request, Response } from 'express';
+
+// npm library for elo calculations
+const EloRating = require('elo-rating');
+
+const User = require('../../models/userModel');
+const BattleRecord = require('../../models/battleRecordModel');
+const Tank = require('../../models/tankModel');
+
+exports.prepareMatch = async (req: Request, res: Response) => {
+
+    // Contains errors for failed validation.
+   const errors = validationResult(req);
+
+   if (!errors.isEmpty()) {
+       // Return 400 for a bad request
+       return res
+           .status(400)
+           .json({ errors: errors.array() });
+   }
+   const { personBeingChallengedId, challengerTankId } = req.body;
+
+   try {
+    // Need to do query explicityly so that the returned doc isn't local to the try catch
```

Comment:
explicityly?

Reply:
Yeah thats what they call it when there's no callback kind an unnecessary comment